### PR TITLE
Generate-docs - need to add indentation to a table if it's a part of a list item.

### PR DESCRIPTION
**Pull request by daryakoval**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/30814

## Description
When generating a README.md file, the numbers in a list are being reset after a list item that contains a table.

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by daryakoval** [comment]
bad

**Comment by daryakoval** [comment]
good

**Comment by ShahafBenYakir** [comment]
The second image is after the fix right?

File: demisto_sdk/commands/generate_docs/common.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def generate_list_section(title, data='', horizontal_rule=False, empty_message='
     return section
 
 
-def generate_table_section(data, title, empty_message='', text='', horizontal_rule=True):
+def generate_table_section(data, title, empty_message='', text='', horizontal_rule=True, numbered_section=False):
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/generate_docs/common.py:99]: Let's add type hints here

**Comment by daryakoval** [comment]
> The second image is after the fix right?

@ShahafBenYakir yes

File: demisto_sdk/commands/generate_docs/common.py (hunk 1 above)
Comment (ShahafBenYakir) [reply demisto_sdk/commands/generate_docs/common.py:99]: And also add this param to doctring

File: demisto_sdk/commands/generate_docs/tests/generate_docs_test.py [hunk 2]
```diff
@@ -508,3 +508,54 @@ def test_generate_integration_doc(self):
...
+2. Search for Zoom.
+3. Click **Add instance** to create and configure a new integration instance.
+
+    | **Parameter** | **Description** | **Required** |
+    | --- | --- | --- |
+    | apiKey |  | True |
+    | apiSecret |  | True |
+    | proxy | Use system proxy settings | False |
+
+4. Click **Test** to validate the URLs, token, and connection."""
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/generate_docs/tests/generate_docs_test.py:540]: Why this is not part of the `fake_README.md`?

File: demisto_sdk/commands/generate_docs/tests/generate_docs_test.py [hunk 3]
```diff
@@ -508,3 +508,54 @@ def test_generate_integration_doc(self):
...
+def test_generate_table_section_numbered_section():
+    """Unit test
+        Given
+            - generate_table_section command
+            - table, that part of numbered section.
+        When
+            - generating setup section of integration
+        Then
+            - Validate that the table has \t at all lines.
+        """
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/generate_docs/tests/generate_docs_test.py:554]:
```suggestion
    """Unit test
        Given
            - A table that should be part of a numbered section (like the setup section of integration README).
        When
            - Running the generate_table_section command.
        Then
            - Validate that the generated table has \t at the beginning of each line.
        """
```

File: demisto_sdk/commands/generate_docs/tests/generate_docs_test.py (hunk 2 above)
Comment (daryakoval) [reply demisto_sdk/commands/generate_docs/tests/generate_docs_test.py:540]: deleted

File: demisto_sdk/commands/generate_docs/common.py (hunk 1 above)
Comment (daryakoval) [reply demisto_sdk/commands/generate_docs/common.py]: done

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,7 @@
 * For partner supported content packs, added support for a list of emails.
 * Removed validation of README files from the ***validate*** command.
 * Fixed an issue where the ***validate*** command required release notes for ApiModules pack.
+* Fixed an issue where the ***generate-docs*** command resets the numbers of the list, after table.
```
Comment (ShahafBenYakir) [inline CHANGELOG.md:14]:
```suggestion
* Fixed an issue where the ***generate-docs*** command reset the enumeration of line numbering after an MD table.
```

File: demisto_sdk/commands/generate_docs/tests/generate_docs_test.py [hunk 5]
```diff
@@ -498,13 +498,41 @@ def teardown_class(cls):
...
             os.path.join(os.path.dirname(TEST_INTEGRATION_PATH), 'README.md')).read()
 
-        # Test that the predefined values and default values are added to the README.
         assert "The type of the newly created user. Possible values are: Basic, Pro, Corporate. Default is Basic." \
                in open(fake_readme).read()
         assert "Number of users to return. Max 300. Default is 30." in open(fake_readme).read()
+
+
+def test_generate_table_section_numbered_section():
+    """Unit test
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/generate_docs/tests/generate_docs_test.py:524]:
```suggestion
```

File: demisto_sdk/commands/generate_docs/tests/generate_docs_test.py [hunk 6]
```diff
@@ -498,13 +498,41 @@ def teardown_class(cls):
...
+        """
+        Given
+            - generate_table_section command
+            - integration
+        When
+            - running generate_integration_doc command on the integration, generating setup section
+        Then
+            - Validate that the generating setup was created correctly, the numbers in the list are not being reset,
+             after a list item that contains a table
+            - Test that the predefined values and default values are added to the README.
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/generate_docs/tests/generate_docs_test.py:510]:
```suggestion
            - YML file representing an integration.
        When
            - Running generate_integration_doc command on the integration.
        Then
            - Validate that the generating setup was created correctly, specifically that the numbers in the lines are not being reset after a list item that contains a table.
            - Test that the predefined values and default values are added to the README.
```
Why did you write in the when section `generating setup section`?

Comment (daryakoval) [reply demisto_sdk/commands/generate_docs/tests/generate_docs_test.py:510]: copy paste mistake

File: demisto_sdk/commands/generate_docs/tests/generate_docs_test.py [hunk 7]
```diff
@@ -498,13 +498,42 @@ def teardown_class(cls):
...
 
     def test_generate_integration_doc(self):
+        """
+        Given
+            - YML file representing an integration.
+        When
+            - Running generate_integration_doc command on the integration.
+        Then
+            - Validate that the generating setup was created correctly, specifically that the numbers in the
+            lines are not being reset after a list item that contains a table.
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/generate_docs/tests/generate_docs_test.py:508]:
```suggestion
            - Validate that the integration README was created correctly, specifically that line numbers are not being reset after a table.
```

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**